c/httpfileservice.c

```diff
@@ -160,7 +148,8 @@ void deleteUnixDirectoryAndRespond(HttpResponse *response, char *absolutePath) {
     response200WithMessage(response, "Successfully deleted a directory");
   }
   else {
-    respondWithJsonError(response, "Failed to delete a directory", 500, "Internal Server Error");
+    respondWithJsonError(response, "Failed to delete a directory. This is usually "
+	"because of permission problems", 403, "Forbidden");
```

Comment:
403 may be misleading here. If someone only checks for http codes, they'll think there is really a permission problem whereas the file can be just missing.

Reply:
If the file is missing, we would get a 404 prior to reaching this point. 
(see https://github.com/zowe/zss/blob/5638836e4e773a45c977a40480f0526ab3d09b90/c/unixFileService.c#L714)
You can view Jordan and I's email chain for info, but here are the things we know when we get to this point in the code.

1. The server is running
2. The user is authenticated
3. The path is either a file or folder and
4. The file or folder exists, and can be proven to exist from the perspective of the server

The question is, are these conditions suffice to determine that if we can't delete the file or folder at this point, we are probably blocked by permissions or something else related.

Reply:
> The file or folder exists, and can be proven to exist from the perspective of the server

No, ```deleteUnixDirectory``` may fail due to other reasons, for example, if the file/directory doesn't exist. A more correct way would be mapping the return codes from the system services we use to the corresponding response message texts (permission denied, file missing, etc).

Reply:
@ifakhrutdinov That's not true, because `deleteUnixDirectory` gets called in `unixfileservice.c`, and thus if the file/directory doesn't exist, it would throw a 404 _before_ reaching this function in `httpfileservice.c` at all

So those 4 conditions _are_ true in the context of how they are used, however if the functions get called by themselves then they're not true. So I think what I'll do is to make the functions work better if they're called somewhere else to future-proof

1. keep isDir
2. fix isDir to work as a boolean
3. use isDir to check if deletion should happen for a file or dir
4. fix delete methods to do checks for if the file exists

Reply:
Thanks @fkovinAtRocket I'll use 400 instead. I don't want to pile through response codes and texts right now, I have Editor work to finish.